### ISSUE TITLE
imgtool listformats segfault

### ISSUE THREAD

**Issue by n6il** [mamedev/mame#7906]
Imgtool seems to just segfault when trying to list the formats. 

```
$ ./imgtool listformats

Segmentation fault: 11
```
 I have no idea how the assertion passed to get to this failure.
 
 ```
 $ lldb ./imgtool
(lldb) target create "./imgtool"
Current executable set to '/Users/mfurman/src/mame/imgtool' (x86_64).
(lldb) run listformats
Process 5101 launched: '/Users/mfurman/src/mame/imgtool' (x86_64)

imgtool was compiled with optimization - stepping may behave oddly; variables may not be available.
Process 5101 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x20)
    frame #0: 0x0000000100004e72 imgtool`imgtool_floppy_make_class(index=0, imgclass=0x00007ffeefbff740) at iflopimg.cpp:313:7 [opt]
   310 		format = (const struct FloppyFormat *)
   311 			imgtool_get_info_ptr(imgclass, IMGTOOLINFO_PTR_FLOPPY_FORMAT);
   312 		assert(format);
-> 313 		if (!format[index].construct)
   314 			return false;
   315 	
   316 		imgclass->derived_get_info = imgclass->get_info;
Target 0: (imgtool) stopped.
(lldb) p index
(int) $0 = 0
(lldb) p format
(const FloppyFormat *) $1 = 0x0000000000000000
```

Info: 
```
$ ./mame -version
0.229 (mame0229-371-gd3062ef1088-dirty)
```
macOS Catalina 10.15.7 (19H2)

```
$ clang --version
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

**Comment by ajrhacker** [comment]
Commit 674bfc96bdd2955f108d3a12edae8638731d796b is at fault here.

**Comment by ajrhacker** [comment]
Fixed (at least for now) in 4d3afe00fcda8061ef7ced47daa391cc0bcf9689. @galibert was a bit hasty in removing the legacy PC floppy format; imgtool actually depends on these things.

**Comment by galibert** [comment]
Legacy will be removed hopefully by this summer.  If there is a conflict
between removing legacy and imgtool, imgtool will go away.  Just to be
clear.

  OG.
